### PR TITLE
Chore: Improved logging & event quality across all controllers.

### PR DESCRIPTION
**Pull request by Ealianis**
Quality control on log standards:

- Uses KObjRef for structured logging 
- Standardizing string messages as much as possible. Generic messages where context (resource type) is available via some other field.
- Logs out Namespace + name of resources (including Work resources)
  - This is necessary as the namespace of Work will not be "default" in prod.
- Renamed local variables to match usage e.g. ManifestObj in lieu of WorkObj. So as not to confuse Work resource.

Fixes #
I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/work_status_controller.go [hunk 1]
```diff
@@ -35,13 +37,11 @@ import (
...
 	workapi "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
+	"sigs.k8s.io/work-api/pkg/utils"
 )
 
 const (
-	EventReasonResourceUnsuccessfullyGarbageCollected = "ResourceUnsuccessfullyGarbageCollected"
-	EventReasonResourceSuccessfullyGarbageCollected   = "ResourceSuccessfullyGarbageCollected"
-	EventReasonAppliedWorkUnsuccessfulUpdated         = "AppliedWorkUnsuccessfulUpdated"
-	EventReasonAppliedWorkSuccessfulUpdated           = "AppliedWorkSuccessfulUpdated"
+	messageWorkStatusReconcileTriggered = "Work Status controller reconcile loop triggered"
```
Comment (ryanzhang-oss) [inline pkg/controllers/work_status_controller.go:44]: why add this var?

File: pkg/controllers/work_status_controller.go [hunk 2]
```diff
@@ -75,29 +76,30 @@ func (r *WorkStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
 	if err = r.spokeClient.Status().Update(ctx, appliedWork, &client.UpdateOptions{}); err != nil {
-		klog.ErrorS(err, "update appliedWork status failed", "appliedWork", appliedWork.GetName())
-		r.recorder.Eventf(work, v1.EventTypeWarning, EventReasonAppliedWorkUnsuccessfulUpdated, "Could not update AppliedWork %s", appliedWork.GetName())
+		klog.ErrorS(err, utils.MessageResourceStatusUpdateFailed, appliedWork.Kind, appliedWork.GetName())
+		r.recorder.Event(appliedWork, v1.EventTypeWarning, utils.EventReasonResourceStatusUpdateFailed, utils.MessageResourceStatusUpdateFailed)
 		return ctrl.Result{}, err
 	}
 
-	r.recorder.Eventf(work, v1.EventTypeNormal, EventReasonAppliedWorkSuccessfulUpdated, "Updated AppliedWork %s", appliedWork.GetName())
+	r.recorder.Event(appliedWork, v1.EventTypeNormal, utils.EventReasonResourceUpdateStatusSucceeded, utils.MessageResourceStatusUpdateSucceeded)
```
Comment (ryanzhang-oss) [inline pkg/controllers/work_status_controller.go:102]: I don't see a reason for this event, event is not log.

File: pkg/controllers/apply_controller.go [hunk 3]
```diff
@@ -43,18 +43,12 @@ import (
...
-	workFieldManagerName                 = "work-api-agent"
-	eventReasonNewManifestApplied        = "NewManifestApplied"
-	eventReasonReconciliationAggregated  = "WorkReconciliationAggregated"
-	eventReasonResourceNotFound          = "ResourceNotFound"
-	eventReasonResourceNotOwnedByWorkAPI = "ResourceNotOwnedByWorkAPI"
-	eventReasonUpdateWorkStatusFailed    = "UpdateWorkStatusFailed"
-	eventReasonUpdateWorkStatusSuccess   = "UpdateWorkStatusSuccessful"
-	eventReasonWorkResourcePatched       = "WorkResourcePatched"
-	eventReasonWorkResourcePatchFailed   = "WorkResourcePatchFailed"
+	messageWorkApplyReconcileTriggered = "Work apply controller reconcile loop triggered"
```
Comment (ryanzhang-oss) [inline pkg/controllers/apply_controller.go:50]: this is only used once, right?

File: pkg/controllers/apply_controller.go [hunk 4]
```diff
@@ -86,26 +81,34 @@ func (r *ApplyWorkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+	// We created the AppliedWork before setting the finalizer, it should already exist.
+	// The AppliedWork.Name shall match its respective Work resource.
 	appliedWork := &workv1alpha1.AppliedWork{}
-	if err := r.spokeClient.Get(ctx, types.NamespacedName{Name: req.Name}, appliedWork); err != nil {
-		klog.ErrorS(err, "failed to get the appliedWork", "name", req.Name)
-		r.recorder.Eventf(work, v1.EventTypeWarning, eventReasonResourceNotFound, "AppliedWork not found for Work %s", work.GetName())
-		return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("failed to get the appliedWork %s", req.Name))
+	if err := r.spokeClient.Get(ctx, types.NamespacedName{Name: kLogObjRef.Name}, appliedWork); err != nil {
+		klog.ErrorS(err, utils.MessageResourceRetrieveFailed, "AppliedWork", kLogObjRef.Name)
+		r.recorder.Eventf(
```
Comment (ryanzhang-oss) [inline pkg/controllers/apply_controller.go:104]: event recorder is not log

File: pkg/controllers/apply_controller.go [hunk 5]
```diff
@@ -145,125 +150,160 @@ func (r *ApplyWorkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
-		Get(ctx, workObj.GetName(), metav1.GetOptions{})
+		Namespace(manifestObj.GetNamespace()).
+		Get(ctx, manifestObj.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
-		actual, err := r.spokeDynamicClient.Resource(gvr).Namespace(workObj.GetNamespace()).Create(
-			ctx, workObj, metav1.CreateOptions{FieldManager: workFieldManagerName})
-		r.recorder.Eventf(workObj, v1.EventTypeNormal, eventReasonNewManifestApplied, "Manifest Applied for Work %s", workObj.GetName())
+		actual, err := r.spokeDynamicClient.Resource(gvr).Namespace(manifestObj.GetNamespace()).Create(
+			ctx, manifestObj, metav1.CreateOptions{FieldManager: utils.WorkFieldManagerName})
+		r.recorder.Event(
```
Comment (ryanzhang-oss) [inline pkg/controllers/apply_controller.go:248]: definitely not for each manifest

File: pkg/controllers/apply_controller.go [hunk 6]
```diff
@@ -145,125 +150,160 @@ func (r *ApplyWorkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+				v1.EventTypeWarning,
+				utils.EventReasonResourcePatchFailed,
+				utils.MessageResourcePatchFailed)
 
 			return nil, false, err
 		}
-		klog.V(5).InfoS("work object patched", "gvr", gvr, "obj", workObj.GetName())
-		r.recorder.Eventf(workObj, v1.EventTypeWarning, eventReasonWorkResourcePatched, "work resource patch successful for %s", workObj.GetName())
+		klog.V(5).InfoS(utils.MessageResourcePatchSucceeded, "gvr", gvr, "obj", kLogObjRef)
+		r.recorder.Event(
```
Comment (ryanzhang-oss) [inline pkg/controllers/apply_controller.go:302]: no manifest event

**Comment by ryanzhang-oss** [comment]
I think the more important task is to make sure that all the work functionalities are there

File: pkg/controllers/work_status_controller.go (hunk 1 above)
Comment (Ealianis) [reply pkg/controllers/work_status_controller.go:44]: It's a style choice to remove string literals from the methods. I can add back if you prefer.

File: pkg/controllers/apply_controller.go (hunk 3 above)
Comment (Ealianis) [reply pkg/controllers/apply_controller.go:50]: Yes, it is a style choice, the variable is private. It's about removing string literals from methods.  I'll remove as  this isn't your desired style.

File: pkg/controllers/apply_controller.go (hunk 4 above)
Comment (Ealianis) [reply pkg/controllers/apply_controller.go:104]: Yup, events are state change. Removing.

File: pkg/controllers/apply_controller.go (hunk 5 above)
Comment (Ealianis) [reply pkg/controllers/apply_controller.go:248]: @ryanzhang-oss , you want to have a single event for if all manifests were applied successfully?

Comment (Ealianis) [reply pkg/controllers/apply_controller.go:248]: Base on your other comments, not for each manifest, but not any manifest event.

File: pkg/controllers/apply_controller.go (hunk 6 above)
Comment (Ealianis) [reply pkg/controllers/apply_controller.go:302]: Removing

File: pkg/controllers/work_status_controller.go (hunk 2 above)
Comment (Ealianis) [reply pkg/controllers/work_status_controller.go:102]: Removed

**Review by aagusuab — APPROVED** [review]

**Pull request merged**